### ISSUE TITLE
Breaks on superfence blocks with `*`

### ISSUE THREAD

**Issue by EvaSDK** [KyleKing/mdformat-mkdocs#21]
When installing mdformat-mkdocs, valid Markdown starts failing to format with mdformat with the following output:

```console
$ mdformat ./test.md
Error: Could not format "<redacted_path>/test.md".

The formatted Markdown renders to different HTML than the input Markdown. This
is likely a bug in mdformat. Please create an issue report here, including the
```

Sample `test.md`:
```text
# My doc

1. Item 1

   ```text
   * sample plain text block.
   ```
```
(non-breakable space used in sample to preserve formatting)

if the fence block is un-indented, mdformat works. If the `*` is replaced with a something else like `-`, mdformat works as well.
This problem manifested itself after copy-pasting some shell tool output which is using `*`. It appears mdformat-mkdocs is trying to convert it to a list.

**Comment by EvaSDK** [comment]
This is probably ugly but I made it work with this:

```diff
diff --git a/mdformat_mkdocs/_normalize_list.py b/mdformat_mkdocs/_normalize_list.py
index 1e7634b..486bdfc 100644
--- a/mdformat_mkdocs/_normalize_list.py
+++ b/mdformat_mkdocs/_normalize_list.py
@@ -122,9 +122,23 @@ def parse_line(line_num: int, content: str) -> ParsedLine:

 def acc_line_results(parsed_lines: list[ParsedLine]) -> list[LineResult]:
     results: list[LineResult] = []
+    fenced_block = False
+
     for parsed in parsed_lines:
         parent_idx = 0
         parents = []
+
+        if fenced_block and parsed.syntax != Syntax.EDGE_CODE:
+            parsed = ParsedLine(
+                line_num=parsed.line_num,
+                indent=parsed.indent,
+                content=parsed.content,
+                syntax=None,
+            )
+
+        if parsed.syntax == Syntax.EDGE_CODE:
+            fenced_block = not fenced_block
+
         with suppress(StopIteration):
             parent_idx, parent = next(
                 (len(results) - idx, line)
```

**Comment by KyleKing** [comment]
Thanks for using `mdformat-mkdocs` and debugging this issue!

This is a busy week for me, but I may have some time this afternoon and will try to release a patch

**Comment by KyleKing** [comment]
I just pushed a fix in v2.0.8. Let me know if you run into any issues!

**Comment by EvaSDK** [comment]
2.0.8+ works fine, thanks.
